API, Core, Flink, Parquet, Spark: Use enhanced for loop

krvikash opened the pull request:
API, Core, Flink, Parquet, Spark: Use enhanced for loop

hililiwei commented on core/src/jmh/java/org/apache/iceberg/util/ZOrderByteUtilsBenchmark.java:
```diff
@@ -114,9 +113,8 @@ public void interleaveValuesFourColumns8ByteOutput(Blackhole blackhole) {
-    for (int i = 0; i < fourColumnInput.length; i++) {
-      byte[] interleavedBytes =
-          ZOrderByteUtils.interleaveBits(fourColumnInput[i], outputSize, outputBuffer);
+    for (byte[][] bytes : fourColumnInput) {
```
bytes -> columnsBinary

hililiwei commented on core/src/jmh/java/org/apache/iceberg/util/ZOrderByteUtilsBenchmark.java:
```diff
@@ -75,9 +75,8 @@ public void interleaveValuesFourColumns(Blackhole blackhole) {
-    for (int i = 0; i < fourColumnInput.length; i++) {
-      byte[] interleavedBytes =
-          ZOrderByteUtils.interleaveBits(fourColumnInput[i], outputSize, outputBuffer);
+    for (byte[][] bytes : fourColumnInput) {
```
bytes -> columnsBinary

hililiwei reviewed (commented):
Good idea. The for-each loop gets rid of the clutter and the opportunity for error by hiding the iterator or index variable.

hililiwei reviewed (commented):
Good idea. The for-each loop gets rid of the clutter and the opportunity for error by hiding the iterator or index variable.

krvikash commented:
> Good idea. The for-each loop gets rid of the clutter and the opportunity for error by hiding the iterator or index variable.

Thanks, @hililiwei for reviewing. Addressed the comments.

amogh-jahagirdar reviewed (approved):
Agreed with @hililiwei, this improves the readability. Thanks for the contribution!

szehon-ho commented:
Merged, thanks @krvikash , @hililiwei , @amogh-jahagirdar
